advanced/matrix/test_matrix.py

```diff
@@ -0,0 +1,148 @@
+"""Unit testing for the Matrix class."""
+import unittest
+
+from matrix import Matrix
+
+
+class TestMatrix(unittest.TestCase):
+    """Test the initialization and methods of the matrix"""
+
+    def setUp(self):
+        """Initialize matrices a and b before tests."""
+        self.mat_a = Matrix(((1, 2), (3, 4)))
+        self.mat_b = Matrix(((3, 5), (6, 8)))
+
+    def test_tuples_property(self):
+        """Test tuples property of matrix."""
+        self.assertEqual(self.mat_a.tuples, ((1, 2), (3, 4)))
+
+    def test_type_validation(self):
+        """Test that only a tuple of tuples is allowed in Matrix."""
+        with self.assertRaises(TypeError):
+            Matrix(((1, 2, 3), [4, 5, 6], 5))
+
+    def test_value_validation(self):
+        """Test that only n*n matrices are allowed."""
+        with self.assertRaises(ValueError):
+            Matrix(((1, 2, 3), (1, 2, 3)))
+
+    def test_unity_2_dimensional(self):
+        """Test unity method on 2 by 2 matrix."""
+        self.assertEqual(Matrix.unity(2), Matrix(((1, 0), (0, 1))))
+
+    def test_unity_3_dimensional(self):
+        """Test unity method on 3 by 3 matrix."""
+        self.assertEqual(Matrix.unity(3),
+                         Matrix(((1, 0, 0), (0, 1, 0), (0, 0, 1))))
```

Comment:
You may missed it, but you checked here two functions:
1. unity function
2. __eq__ function
Generally it is ok to use multiple functions in single test, but you need to check they are working before using them (the functions that are not the purpose of the test). 
In this case the test might have failed, but because of the __eq__ and not because the unity
(This is correct for all tests that do it)